### PR TITLE
`bd_rhapsody_make_reference`: Create a reference for the BD Rhapsody pipeline

### PR DESCRIPTION
**Pull request by rcannood**
### NEW FEATURES

* `bd_rhapsody`:
  - `bd_rhapsody/bd_rhapsody_make_reference`: Create a reference for the BD Rhapsody pipeline (PR #xx).

## Issue ticket number

Closes #xxxx 

## Checklist before requesting a review

- [x] I have performed a self-review of my code

- [x] Conforms to the [Contributing guidelines](https://github.com/viash-hub/base/blob/main/CONTRIBUTING.md)

- [x] Proposed changes are described in the [CHANGELOG.md](https://github.com/viash-hub/base/blob/main/CHANGELOG.md)

- [x] I have tested my code with `viash ns test --parallel -q <name or namespace>`

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Documentation
  - [ ] Bug fixes

### CODE REVIEW

File: src/bd_rhapsody/bd_rhapsody_make_reference/test.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -e
```
Comment (DriesSchaumont) [inline src/bd_rhapsody/bd_rhapsody_make_reference/test.sh:3]: Add `pipefail`

File: src/bd_rhapsody/bd_rhapsody_make_reference/test_data/script.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    -O $ORIG_GTF
+fi
+
+# create small reference
+START=30000
+END=31500
+CHR=chr1
+
+# subset to small region
+seqkit grep -r -p "^$CHR\$" "$ORIG_FA" | \
```
Comment (DriesSchaumont) [inline src/bd_rhapsody/bd_rhapsody_make_reference/test_data/script.sh:32]: Add a check at the beginning of the script that checks if seqkit is installed.

**Review by DriesSchaumont — CHANGES_REQUESTED** [review]

File: src/bd_rhapsody/bd_rhapsody_make_reference/config.vsh.yaml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+license: Unknown
+requirements:
+  commands: [ "cwl-runner" ]
+argument_groups:
+  - name: Inputs
+    arguments:
+      - type: file
+        name: --genome_fasta
+        required: true
+        description: Reference genome file in FASTA or FASTA.GZ format. The BD Rhapsody Sequencing Analysis Pipeline uses GRCh38 for Human and GRCm39 for Mouse.
```
Comment (DriesSchaumont) [inline src/bd_rhapsody/bd_rhapsody_make_reference/config.vsh.yaml:25]:
```suggestion
        description: Reference genome file in FASTA or FASTA.GZ format.
```

File: src/bd_rhapsody/bd_rhapsody_make_reference/config.vsh.yaml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      - type: file
+        name: --gtf
+        required: true
+        description: |
+          File path to the transcript annotation files in GTF or GTF.GZ format. The Sequence Analysis Pipeline requires the 'gene_name' or 
+          'gene_id' attribute to be set on each gene and exon feature. Gene and exon feature lines must have the same attribute, and exons
+          must have a corresponding gene with the same value. For TCR/BCR assays, the TCR or BCR gene segments must have the 'gene_type' or
+          'gene_biotype' attribute set, and the value should begin with 'TR' or 'IG', respectively. The BD Rhapsody Sequencing Analysis
+          Pipeline uses Gencode v42 for Human and M31 for Mouse.
+        example: transcriptome_annotation.gtf.gz
```
Comment (DriesSchaumont) [inline src/bd_rhapsody/bd_rhapsody_make_reference/config.vsh.yaml:39]:
```suggestion
          'gene_biotype' attribute set, and the value should begin with 'TR' or 'IG', respectively.
        example: transcriptome_annotation.gtf.gz
```

**Review by DriesSchaumont — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,10 @@
 # biobox x.x.x
 
+## NEW FEATURES
+
+* `bd_rhapsody`:
+  - `bd_rhapsody/bd_rhapsody_make_reference`: Create a reference for the BD Rhapsody pipeline (PR #xx).
```
[the remark this answers is not in the document]
Comment (rcannood) [reply CHANGELOG.md:6]: Done!

File: src/bd_rhapsody/bd_rhapsody_make_reference/config.vsh.yaml [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+            If you have already pre-filtered the input Annotation files and/or wish to turn-off the filtering, please set this option to True.
+        info:
+          config_key: Filtering_off
+      - type: boolean_true
+        name: --wta_only_index
+        description: Build a WTA only index, otherwise builds a WTA + ATAC index.
+        info:
+          config_key: Wta_Only
+      - type: string
```
[the remark this answers is not in the document]
Comment (rcannood) [reply src/bd_rhapsody/bd_rhapsody_make_reference/config.vsh.yaml:105]: It isn't

File: src/bd_rhapsody/bd_rhapsody_make_reference/test_data/script.sh (hunk 2 above)
Comment (rcannood) [reply src/bd_rhapsody/bd_rhapsody_make_reference/test_data/script.sh:32]: done!

Comment (rcannood) [reply src/bd_rhapsody/bd_rhapsody_make_reference/test_data/script.sh:32]: Done!

File: src/bd_rhapsody/bd_rhapsody_make_reference/config.vsh.yaml [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+keywords: [genome, reference, index, align]
+links:
+  repository: https://bitbucket.org/CRSwDev/cwl/src/master/v2.2.1/Extra_Utilities/
+  documentation: https://bd-rhapsody-bioinfo-docs.genomics.bd.com/resources/extra_utilities.html#make-rhapsody-reference
+license: Unknown
+authors:
+  - __merge__: /src/_authors/robrecht_cannoodt.yaml
+    roles: [ author, maintainer ]
+  - __merge__: /src/_authors/weiwei_schultz.yaml
+    roles: [ author ]
```
Comment (dorien-er) [inline src/bd_rhapsody/bd_rhapsody_make_reference/config.vsh.yaml:21]:
```suggestion
    roles: [ contributor ]
```

File: src/bd_rhapsody/bd_rhapsody_make_reference/config.vsh.yaml [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+        required: false
+        multiple: true
+        default: [chrM, chrMT, M, MT]
+        info:
+          config_key: Mitochondrial_contigs
+      - type: boolean_true
+        name: --filtering_off
+        description: |
+          By default the input Transcript Annotation files are filtered based on the gene_type/gene_biotype attribute. Only features 
+          having the following attribute values are are kept:
```
Comment (dorien-er) [inline src/bd_rhapsody/bd_rhapsody_make_reference/config.vsh.yaml:81]:
```suggestion
          having the following attribute values are kept:
```

**Review by dorien-er — CHANGES_REQUESTED** [review]

**Review by dorien-er — APPROVED** [review]

**Pull request merged**